utils.py

```diff
@@ -2125,47 +2135,37 @@ def num_denom(A, symbolic_expr=False):
         denom = [numpy.poly1d(denom) *
                  numpy.poly1d(A.matrix[0, j].denominator.coeffs)
                  for j in range(A.matrix.shape[1])]
-        num = [numpy.poly1d(num) *
-               numpy.poly1d(A.matrix[0, j].numerator.coeffs)
+        num = [numpy.poly1d(num)
+               * numpy.poly1d(A.matrix[0, j].numerator.coeffs)
                for j in range(A.matrix.shape[1])]
-        if symbolic_expr is True:
-            for n in range(len(denom)):
-                sym_den = (sym_den + denom[- n - 1] * s**n).simplify()
-            for n in range(len(num)):
-                sym_num = (sym_num + num[- n - 1] * s**n).simplify()
-            return sym_num, sym_den
-        else:
-            return num, denom
-
     elif type(A) == tf:
-        denom = []
-        num = []
 
         denom = [list(A.denominator.coeffs)[n] for n in range(
             len(list(A.denominator.coeffs)))]
         num = [list(A.numerator.coeffs)[n] for n in range(
             len(list(A.numerator.coeffs)))]
-        if symbolic_expr is True:
-            for n in range(len(denom)):
-                sym_den = (sym_den + denom[- n - 1] * s**n).simplify()
-            for n in range(len(num)):
-                sym_num = (sym_num + num[- n - 1] * s**n).simplify()
-            return sym_num, sym_den
-        else:
-            return num, denom
-"""
+
+    if symbolic_expr is True:
+        for n in range(len(denom)):
```

Comment:
`for i in range(len(` is a code smell in Python. See [this presentation](https://nedbatchelder.com/text/iter.html) for why.

Reply:
I didn't address this because I didn't quite follow what the code was doing, but I'll take another look and see if I can refactor.

Reply:
The branch of the `if` statement for the tf object was easy to fix once I understood what the code was _intended_ to do.

The branch for the mimotf object actually only worked when the function was called with symbolic_expr == False, I have written a very ugly patch to this. I plan to rewrite the mimotf class when I get to that section in the textbook and will come back to this piece of code when that is fixed. The problem is that indexing doesn't work as expected, I believe.